handlers/event-handler.ts

```diff
@@ -21,9 +21,16 @@ export interface EventHandlerDefinition extends HandlerDefinition {
 
 export type EventHandlerEvent<T extends EventBridgeEvent<string, any>> = T;
 
-export interface EventHandlerOptions<T extends EventBridgeEvent<string, any>>
+export interface EventHandlerOptions<T extends string, D extends any>
 	extends EventHandlerDefinition {
-	validator: (detail: EventBridgeEvent<string, any>) => T | void;
+	/** @deprecated use `validators` instead */
+	validator?: (
+		detail: EventBridgeEvent<string, any>,
+	) => EventBridgeEvent<T, D> | void;
+	validators?: {
+		type: (type: string) => T | void;
+		detail: (detail: unknown) => D | void;
+	};
```

Comment:
IMO, I think it would be better for these to be one validator. They're intrinsically linked and one without the other doesn't make too much sense.